lib/ResourceString.js

```diff
@@ -102,6 +102,24 @@ ResourceString.prototype.setTarget = function(str) {
     this.target = str;
 };
 
+/**
+ * Return the resource key written in the target locale.
+ *
+ * @returns {String} the source string
+ */
+ResourceString.prototype.getReskey = function() {
+    return this.reskey;
+};
+
+/**
+ * Set the resource key string of this resource.
+ *
+ * @param {String} str the target string
+ */
+ResourceString.prototype.setReskey = function(str) {
```

Comment:
The key is immutable once the resource is created, so you should not be able to set it.